export const pgException = {
  divisionByZero: "division_by_zero",
  invalidCursorName: "invalid_cursor_name",
  undefinedColumn: "undefined_column",
  undefinedTable: "undefined_table",
  duplicateTable: "duplicate_table",
  duplicateObject: "duplicate_object",
  duplicateColumn: "duplicate_column",
  uniqueViolation: "unique_violation",
  foreignKeyViolation: "foreign_key_violation",
  checkViolation: "check_violation",
  notNullViolation: "not_null_violation",
  invalidTextRepresentation: "invalid_text_representation",
  numericValueOutOfRange: "numeric_value_out_of_range",
  stringDataRightTruncation: "string_data_right_truncation",
  syntaxError: "syntax_error",
  insufficientPrivilege: "insufficient_privilege",
  undefinedFunction: "undefined_function",
  undefinedObject: "undefined_object",
  others: "others",
} as const;

export const pgExceptionMessage = {
  divisionByZero: "Division by zero was attempted",
  invalidCursorName: "Cursor name not found",
  undefinedColumn: "Referenced column does not exist",
  undefinedTable: "Table does not exist",
  duplicateTable: "Table already exists",
  duplicateObject: "Generic object already exists (domain, type, schema, etc.)",
  duplicateColumn: "Column already exists",
  uniqueViolation: "Insert/update violates unique constraint",
  foreignKeyViolation: "Violates a foreign key constraint",
  checkViolation: "Fails a CHECK constraint",
  notNullViolation: "Column is null but defined as NOT NULL",
  invalidTextRepresentation: "Bad input format (e.g. 'abc'::int)",
  numericValueOutOfRange: "Value out of allowed range",
  stringDataRightTruncation: "String too long for column type",
  syntaxError: "SQL syntax problem",
  insufficientPrivilege: "Permission denied",
  undefinedFunction: "Function does not exist",
  undefinedObject: "Generic object not found",
  others: "Unknown error",
} as const;
